src/recovery/mod.rs

```diff
@@ -1041,6 +1043,9 @@ impl std::fmt::Debug for Recovery {
             write!(f, "hystart={:?} ", self.hystart)?;
         }
 
+        // CC-specific debug info
+        write!(f, "{} ", (self.cc_ops.debug_string)(self))?;
```

Comment:
Instead of returning a string, it seems like it would be better to pass the `std::fmt::Formatter` directly to the callback. This way the cb can call `write!()` directly, without having to allocate a temporary string that would then get copied into the formatter and discarded anyway.

Reply:
Updated as recommended. Thanks!